python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -389,10 +393,14 @@ def _resize_channel_if_needed(self, serialized_value: str):
                 special_message_serialized,
                 prev_writer_ref,
                 self._num_readers,
+                timeout_ms,
             )
 
-    def write(self, value: Any):
+    def write(self, value: Any, timeout: Optional[float] = None) -> None:
         self.ensure_registered_as_writer()
+        assert timeout is None or timeout >= 0, "Timeout must be non-negative."
+        # -1 means no timeout (block indefinitely)
+        timeout_ms = int(timeout * 1000) if timeout is not None else -1
```

Comment:
Let's try to be consistent whether we use timeout=None vs timeout=-1 for infinity.

Typically in Python, the `None` value only means "default value", so I think -1 for infinity is the better convention.

Reply:
See my other comment as well: this is to be consistent with ray.get() which we share code path. More thoughts on this? I can change to -1 if that is still a preference.

Reply:
I think ray.get just does that because None means default, -1 means infinity, and the default is infinity. Usually we use None as the default kwarg value so we can distinguish between user wanting the default value vs user wanting to override the default. Using None lets us easily change the default value later if needed.

Reply:
So I think using None as the default kwarg value is good, but let's use -1 to mean infinity